packages/flutter_sprites/lib/src/sound.dart

```diff
@@ -1,139 +1,111 @@
 part of flutter_sprites;
 
-// TODO: The sound effects should probably use Android's SoundPool instead of
-// MediaPlayer as it is more efficient and flexible for playing back sound effects
-
-typedef void SoundEffectStreamCallback(SoundEffectStream stream);
-
+/// An audio asset loaded by the SoundEffectPlayer.
 class SoundEffect {
-  SoundEffect(this._pipeFuture);
-
-  // TODO: Remove load method from SoundEffect
-  Future load() async {
-    _data = await _pipeFuture;
-  }
+  int _soundId;
 
-  Future<MojoDataPipeConsumer> _pipeFuture;
-  MojoDataPipeConsumer _data;
+  SoundEffect(this._soundId);
 }
 
+/// A sound being played by the SoundEffectPlayer.
 class SoundEffectStream {
-  SoundEffectStream(
-    this.sound,
-    this.loop,
-    this.volume,
-    this.pitch,
-    this.pan,
-    this.onSoundComplete
-  );
-
-  // TODO: Make these properties work
-  SoundEffect sound;
-  bool playing = false;
-  bool loop = false;
-  double volume = 1.0;
-  double pitch = 1.0;
-  double pan = 0.0;
-
-  // TODO: Implement completion callback. On completion, sounds should
-  // also be removed from the list of playing sounds.
-  SoundEffectStreamCallback onSoundComplete;
+  int _streamId;
+  bool _paused;
+  bool _loop;
+  double _leftVolume;
+  double _rightVolume;
+  double _pitch;
```

Comment:
all the fields that back getters and setters should be inline with their getters and setters, on the line after their getters.